src/GNDStk/gnds-keyword-meta.hpp

```diff
@@ -0,0 +1,195 @@
+make_meta(bool, identicalParticles);
+make_meta(bool, supportsAngularReconstruction);
+make_meta(bool, useForSelfShieldingOnly);
+
+// int
+make_meta(int, A);
+make_meta(int, columnIndex);
+make_meta(int, columns);
+make_meta(int, ENDF_MT);
+make_meta(int, index);
+make_meta(int, L);
+make_meta(int, MAT);
+make_meta(int, matrixStartIndex);
+make_meta(int, metaStableIndex);
+make_meta(int, nParameters);
+make_meta(int, numberOfProducts);
+make_meta(int, numberPerMolecule);
+make_meta(int, parity);
+make_meta(int, rows);
+make_meta(int, Z);
+
+// double
+make_meta(double, coefficient);
+make_meta(double, domainMax);
+make_meta(double, domainMin);
+make_meta(double, electronNumber);
+make_meta(double, format);
+make_meta(double, max);
+make_meta(double, min);
+make_meta(double, muCutoff);
+
+// string
+make_meta(std::string, approximation);
+make_meta(std::string, asymmetric);
+make_meta(std::string, boundaryCondition);
+make_meta(std::string, complete);
+make_meta(std::string, compression);
+make_meta(std::string, crossTerm);
+make_meta(std::string, dependenceOnProcessedGroupWidth);
+make_meta(std::string, derivedFrom);
+make_meta(std::string, domainUnit);
+make_meta(std::string, ejectile);
+make_meta(std::string, eliminated);
+make_meta(std::string, emissionMode);
+make_meta(std::string, encoding);
+make_meta(std::string, ENDF_MFMT);
+make_meta(std::string, evaluation);
+make_meta(std::string, final);
+make_meta(std::string, fissionGenre);
+make_meta(std::string, flags);
+make_meta(std::string, functionalForm);
+make_meta(std::string, generation);
+make_meta(std::string, genre);
+make_meta(std::string, href);
+make_meta(std::string, id);
+make_meta(std::string, initial);
+make_meta(std::string, interpolation);
+make_meta(std::string, interpolationQualifier);
+make_meta(std::string, label);
+make_meta(std::string, library);
+make_meta(std::string, material);
+make_meta(std::string, mode);
+make_meta(std::string, name);
+make_meta(std::string, path);
+make_meta(std::string, pid);
+make_meta(std::string, process);
+make_meta(std::string, productFrame);
+make_meta(std::string, projectile);
+make_meta(std::string, projectileFrame);
+make_meta(std::string, resonanceReaction);
+make_meta(std::string, shape);
+make_meta(std::string, style);
+make_meta(std::string, subshell);
+make_meta(std::string, symbol);
+make_meta(std::string, symmetry);
+make_meta(std::string, target);
+make_meta(std::string, type);
+make_meta(std::string, unit);
+make_meta(std::string, version);
+make_meta(std::string, valueType);
+make_meta(std::string, date);
+make_meta(std::string, decayRate); // seems to always have double and "1/s"
+
+} // namespace meta
+
+
+
+// -----------------------------------------------------------------------------
+// keywords
+// Special cases, not doable with our macro.
+// -----------------------------------------------------------------------------
+
+// fixme
+// For the variant cases, we should work out a way to make the type be context-
```

Comment:
Generally, upon looking through the GNDS files, it was pretty clear what type of data appears in metadata values or in child nodes with direct content (e.g. parsed character data). There was a handful of cases where it wasn't absolutely clear. I'll have to go check, but I think there was a "degrees of freedom" thing in some GNDS files, that sometimes was an integer and sometimes a real non-integer (mathematically speaking). I suppose I could cover both cases by making it a real, but (without knowing the exact meaning), I'd really have expected an integer for a "degrees of freedom," by most definitions I've seen of the phrase. Perhaps a user would care about whether the thing actually appears as a straight integer in the file, or appears instead as a real. I'm not a big fan of variant, as you know, but this seemed like a case where it might be put to good use.